vslib/src/CorePortIndexMapFileParser.cpp

```diff
@@ -0,0 +1,230 @@
+#include "CorePortIndexMapFileParser.h"
+
+#include "swss/logger.h"
+#include "swss/tokenize.h"
+
+#include <net/if.h>
+
+#include <fstream>
+
+using namespace saivs;
+
+// must be the same or less as SAI_VS_SWITCH_INDEX_MAX
+#define MAX_SWITCH_INDEX (255)
+
+bool CorePortIndexMapFileParser::isInterfaceNameValid(
+        _In_ const std::string& name)
+{
+    SWSS_LOG_ENTER();
+
+    size_t size = name.size();
+
+    if (size == 0 || size > IFNAMSIZ)
+    {
+        SWSS_LOG_ERROR("invalid interface name %s or length: %zu", name.c_str(), size);
+        return false;
+    }
+
+    for (size_t i = 0; i < size; i++)
+    {
+        char c = name[i];
+
+        if (c >= '0' && c <= '9')
+            continue;
+
+        if (c >= 'a' && c <= 'z')
+            continue;
+
+        if (c >= 'A' && c <= 'Z')
+            continue;
+
+        SWSS_LOG_ERROR("invalid character '%c' in interface name %s", c, name.c_str());
+        return false;
+    }
```

Comment:
find helper function in std, like std::is_digit etc

Reply:
Fixed in PR #700